#!/usr/bin/env python
PACKAGE = "unicorn_tracking"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("lambda",       double_t,       0,      "Threshold Clustering",     0.15,     0.0,    1.0 )
gen.add("max_dist_laser",       double_t,       0,      "Max Laser Distance Allowed",    10,     0.0,    100.0 )
gen.add("static_filter",       bool_t,       0,      "Static Map Filter",  True)
gen.add("static_remove_dist",       int_t,       0,      "Static Map Filter Tolerance",  4, 1, 100)
gen.add("static_remove_ratio",       double_t,       0,      "Static Map Filter Ratio required",  0.5, 0.0, 1.0)
gen.add("polygon_tolerance",       double_t,       0,      "Shape Extraction Tolerance",  1.04, 1.0, 2.0)
gen.add("polygon_min_points",       int_t,       0,      "Shape min amount of points",  4, 2, 100)
gen.add("min_twist_detection",       double_t,       0,      "Minimum speed detected",  0.08, 0.0, 10.0)
gen.add("sim_adj_dist",       double_t,       0,      "Similarity Weight: Side Length",  1.0, 0.0, 10.0)
gen.add("sim_adj_angle",       double_t,       0,      "Similarity Weight: Mean angle",  1.0, 0.0, 10.0)
gen.add("sim_adj_side",       double_t,       0,      "Similarity Weight: Amount of Detected Sides",  3.0, 0.0, 10.0)
gen.add("sim_adj_xpos",       double_t,       0,      "Similarity Weight: X position, tracked",  5.0, 0.0, 10.0)
gen.add("sim_adj_ypos",       double_t,       0,      "Similarity Weight: Y position, tracked",  5.0, 0.0, 10.0)
gen.add("sim_adj_posdiff",       double_t,       0,      "Similarity Weight: position compared to previous position",  5.0, 0.0, 10.0)
gen.add("max_similarty_deviation",       double_t,       0,      "Max allowed Similarty Deviation",  1.5, 0.0, 10.0)
gen.add("min_size_cluster",       int_t,       0,      "Min points per cluster",  5, 1, 100)
gen.add("TRACKER_LIFE",       int_t,       0,      "Tracker Life (when out of sight)",  1000, 1, 10000)
gen.add("CONFIRMED_TRACK",       int_t,       0,      "Timer for confirmed track",  100, 1, 10000)




exit(gen.generate(PACKAGE, "unicorn_tracking_node", "Tracking"))